src/mongo/db/storage/wiredtiger/encryption_keydb.cpp

```diff
@@ -413,7 +420,18 @@ void EncryptionKeyDB::clone(EncryptionKeyDB *old) {
 }
 
 void EncryptionKeyDB::store_masterkey() {
-    vaultWriteKey(base64::encode(StringData{(const char*)_masterkey, _key_len}));
+    auto encode_master_key = [this]() {
+        return base64::encode(StringData{(const char*)_masterkey, _key_len});
```

Comment:
Frankly I would not bother with lambda here. Encoding master key is a cheap operation. Moreover delaying this operation has no sense since it will be called anyway in 100% cases (std::logic_error case is excluded by options parser/validator).
So I would just cache encoded value in the local variable.

Reply:
I agree that I over-engineered that. Changed to a simple local variable.